src/htcondor_es/convert_to_json.py

```diff
@@ -1146,18 +1146,38 @@ def commonExitCode(ad):
     """
     Consolidate the exit code values of JobExitCode,
     the  chirped CRAB and WMCore values, and
-    the original condor exit code.
-    JobExitCode and Chirp_CRAB3_Job_ExitCode
-    exists only on analysis jobs.
+    the original condor exit code, according to 
+    the workflow type: production or analysis.
+
+    Input parameter: ad #this is the raw ad from condor
     """
-    return ad.get(
-        "JobExitCode",
-        ad.get(
-            "Chirp_CRAB3_Job_ExitCode",
-            ad.get("Chirp_WMCore_cmsRun_ExitCode", ad.get("ExitCode", 50666)),
-        ),
-    )
+    # If the raw ExitCode in the ad is not present,
+    # condor did not finish, hence we return 50666
+    # for any workflow type.
+    if not "ExitCode" in ad:
+        return 50666
+    
+    condorExitCode = ad.get("ExitCode")
 
+    analysis = (
```

Comment:
Given that it is used in a couple of times, perhaps making an specific function/lambda like `isAnalysisJob` would avoid divergence in the future. Just a suggestion though, it can be done in the future as well.

Reply:
Function created!